Base/usr/share/man/man2/posix_spawn.md

```diff
@@ -0,0 +1,47 @@
+## Name
+
+`posix_spawn` - launch a new process
+
+## Synopsis
+
+```**c++
+#include <spawn.h>
+
+int posix_spawn(pid_t* pid, const char* file, const posix_spawn_file_actions_t*, const posix_spawnattr_t*, char* const argv[], char* const envp[]);
+int posix_spawnp(pid_t* pid, const char* file, const posix_spawn_file_actions_t*, const posix_spawnattr_t*, char* const argv[], char* const envp[]);
+```
+
+## Description
+
+Spawns a new process reading the binary `file`, passing `argv` as arguments to `main()` and setting `envp` as argument.
+
+Places the pid of the new process in `pid`.
+
+If `file` passed to `posix_spawn` is a relative path, it is resolved relative to the current working directory.
+
+If `file` passed to `posix_spawnp` is a relative path, it is resolved by searching `PATH`.
+
+The `posix_spawn_file_actions_t` and `posix_spawnattr_t` arguments may be `nullptr`. If they aren't, see [`posix_spawn_file_actions`(2)](posix_spawn_file_actions_init.md) and [`posix_spawnattr`(2)](posix_spawnattr_init.md) for what they do.
+
+The last entry in `argv` and `envp` has to be `nullptr`.
+
+To make the subprocess use the parent's environment, declare `extern char* environ;` in your program and pass that.
```

Comment:
It's not "subprocess", it's "child process".

This remark belongs in "notes" or something, not in "description".

You don't have to declare `environ`, it's in unistd:

https://github.com/SerenityOS/serenity/blob/6d5bd8c76b1eec679ec6ca8201eef6ec5ff21a9e/Libraries/LibC/unistd.h#L55

And it's a `char**`.

Reply:
I was led astray by https://man7.org/linux/man-pages/man7/environ.7.html which says " (This variable must be declared in the user program, but is declared in the header file <unistd.h> if the _GNU_SOURCE feature test macro is defined.)".

https://pubs.opengroup.org/onlinepubs/9699919799/functions/environ.html doesn't have any of that language, so thanks :)

Reply:
Heh, a bit of a friendly advice: do read Linux and POSIX man pages — they can be helpful at times, — but do not mistake then for the "real thing" (SerenityOS). SerenityOS ≠ Linux && SerenityOS ≠ POSIX.